src/app/customer/register/page.tsx

```diff
@@ -0,0 +1,152 @@
+"use client";
+
+import { useState } from "react";
+import { useRouter } from "next/navigation";
+import { CustomerEntity } from "@/types/customer";
+import { CustomerService } from "@/api/customerApi";
+import Link from "next/link";
+
+export default function RegisterPage() {
+    const router = useRouter();
+    //admite cualquier dato, así no es necesario declarar el customer como DTO
+    const [formData, setFormData] = useState<Omit<CustomerEntity, 'uri'>>({
+        name: "",
+        email: "",
+        password: "",
+        phoneNumber: "",
+    });
+    const [error, setError] = useState<string>("");
+    const [isLoading, setIsLoading] = useState(false);
+
+    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const { name, value } = e.target;
+        setFormData((prev) => ({
+            ...prev,
+            [name]: value,
+        }));
+    };
+
+    const handleSubmit = async (e: React.FormEvent) => {
+        e.preventDefault();
+        setError("");
+        setIsLoading(true);
+
+        try {
+            const response = await fetch("/customer/register", {
```

Comment:
Wrong API endpoint, like in customerApi.ts. Indeed, better to use the method on that class instead of doing the POST here